### PR TITLE
Add friend be

### PR DESCRIPTION
**Pull request by riceE4ter**
Had some comments for parts i was confused about

### CODE REVIEW

File: app/Http/Controllers/AjaxController.php [hunk 1]
```diff
@@ -29,4 +29,31 @@ public function addfriend(Request $req) {
...
+                ->get();
+            if (isset ($req->accepted)) {
+                if ($req->accepted == 'accepted') {
+                    $friendship->status_code = 'accepted';
+                    return response()->json(array("message" => "friend request accepted!"), 200);
+                } else if ($req->accepted == 'denied') {
+                    $friendship->status_code = 'denied';
+                    return response()->json(array("message" => "friend request denied!"), 200);
+                }
+                $friendship->save(); //may not actually save into db
```
Comment (nikoootine) [inline app/Http/Controllers/AjaxController.php]: you should have the return statement after this line or else you wont be able to save

File: app/Http/Controllers/AjaxController.php [hunk 2]
```diff
@@ -29,4 +29,31 @@ public function addfriend(Request $req) {
...
+            }
+            // accepted is a temp name; may change
+            // What is action_username for?
```
Comment (nikoootine) [inline app/Http/Controllers/AjaxController.php]: action_username is the one who initiated the friend request

File: app/Http/Controllers/AjaxController.php [hunk 3]
```diff
@@ -29,4 +29,31 @@ public function addfriend(Request $req) {
...
+    /*
+     * AJAX Request to handle accepting/declining a friend request
+     *
+     */
+    public function processFriendRequest(Request $req) {
+        if ($req->json() && isset ($req->username)) {
+            $friendship = Friend::whereIn('username1', [Auth::user()->username, $req->username])
+                ->whereIn('username2', [Auth::user()->username, $req->username])
+                ->where('status_code', 'pending')
+                ->get();
```
Comment (nikoootine) [inline app/Http/Controllers/AjaxController.php]: The query is wrong. You should be getting the rows where username2 is the Auth::user() and status_code is pending [and action_username is not username2]

The one in square bracket is optional

Comment (itsmattli) [inline app/Http/Controllers/AjaxController.php]: just as a not for clairty.

`action_username` is the person who sent the friend request, as @nikoootine said you need to check `->where('something','!=','something)`

File: app/Http/Controllers/AjaxController.php (hunk 1 above)
Comment (itsmattli) [inline app/Http/Controllers/AjaxController.php]: you need to call save() on $friendship before you return or nothing will be saved.

File: app/Http/Controllers/AjaxController.php (hunk 2 above)
Comment (itsmattli) [inline app/Http/Controllers/AjaxController.php]: see my earlier comment

File: app/Http/Controllers/AjaxController.php [hunk 4]
```diff
@@ -29,4 +29,31 @@ public function addfriend(Request $req) {
...
+            if (isset ($req->accepted)) {
+                if ($req->accepted == 'accepted') {
```
Comment (itsmattli) [inline app/Http/Controllers/AjaxController.php]: we're using the "approved", "pending", "declined"

File: app/Http/Controllers/AjaxController.php [hunk 5]
```diff
@@ -29,4 +29,31 @@ public function addfriend(Request $req) {
...
+                    $friendship->status_code = 'accepted';
+                    $friendship->save();
+                    return response()->json(array("message" => "friend request accepted!"), 200);
+                } else if ($req->accepted == 'denied') {
+                    $friendship->status_code = 'denied';
+                    $friendship->save();
+                    return response()->json(array("message" => "friend request denied!"), 200);
```
Comment (itsmattli) [inline app/Http/Controllers/AjaxController.php]: @nikoootine @vincent-h-lee what convention do we want to do for our AJAX. 200 means ok, but do we want to send back an Error code if the request is denied?

Comment (nikoootine) [inline app/Http/Controllers/AjaxController.php]: The status codes that are sent back into a response are HTTP status codes.  I dont think we should send 500 even though the friend request is denied. An error code 500 usually means there's something wrong with the http protocol

**Pull request merged**